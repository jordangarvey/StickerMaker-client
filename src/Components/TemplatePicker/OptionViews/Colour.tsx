import React, { FC, Fragment } from "react";
import { StyleSheet, TouchableOpacity } from "react-native";

import Categories from "../Categories";
import Colours from "../Colours";


/**
 * Interface for the Colour component
 */
interface IShapeProps {
	/** Callback to be fired whenever a shape option is pressed */
	onPress(category: Categories, option: Colours): void;
	/** The shape this component should display as */
	colour: Colours;

	/** Optional boolean option value if this shape is selected */
	selected?: boolean;
}

/**
 * Component to render a colour button
 */
const Shape: FC<IShapeProps> = (props) => (
	<Fragment>
		<TouchableOpacity
			onPress={props.onPress.bind(null, Categories.Colour, props.colour)}
			style={
				[
					{
						backgroundColor: props.colour
					},
					styles.colour,
					props.selected && styles.selected
				]
			}
		/>
	</Fragment>
);

const styles = StyleSheet.create({
	colour: {
		borderRadius: 25,
		height: 50,
		width: 50
	},
	selected: {
		borderColor: "green",
		borderWidth: 3
	}
});

export default Shape;
