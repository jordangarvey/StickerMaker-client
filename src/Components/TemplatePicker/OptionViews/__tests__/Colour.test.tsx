import React from "react";

import renderer from "react-test-renderer";

import Colour from "../Colour";


describe("The Colour componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<Colour
				colour="Grey1"
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
