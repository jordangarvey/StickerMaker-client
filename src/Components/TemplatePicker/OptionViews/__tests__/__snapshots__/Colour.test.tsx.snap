// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Colour componentâ€™s snapshot renders as expected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#333333",
      "borderRadius": 30,
      "height": 60,
      "marginHorizontal": 15,
      "opacity": 1,
      "width": 60,
    }
  }
/>
`;
